### PR TITLE
Add support for Vivado HLS generated IPs and float interfaces

### PR DESCRIPTION
**Pull request by davide-giri**
Improvements to the Vivado HLS accelerator design and integration flow:
* commit 90c8e91c7fbd80e2e58f0cbcf4be08703671af00: add automatic integration of IPs generated by Vivado HLS
* commit dde8b86557e40fa0420dc5dc58fbc932f5211f01: add support for Vivado HLS accelerators with float interfaces

This PR resolves issue #44.

### CODE REVIEW

File: utils/sldgen/sld_generate.py [hunk 1]
```diff
@@ -1542,19 +1561,28 @@ def gen_tile_acc(accelerator_list, axi_acceleratorlist, template_dir, out_dir):
...
       if re.match(r'dma[1-9]+', item, re.M|re.I):
         dp_dma_width = int(item.replace("dma", ""))
         if dp_dma_width != dma_width:
           skip = True
           break;
+      if re.match(r'flin', item, re.M|re.I):
+        datatype = "float_in"
+      elif re.match(r'flout', item, re.M|re.I):
+        datatype = "float_out"
+      elif re.match(r'fl', item, re.M|re.I):
```
Comment (paulmnt) [inline utils/sldgen/sld_generate.py:1575]: These regexp are easy to match with other words. We should define something similar to `dma[1-9]+` above.
E.g. fl32in, fl32out, fl32.
I understand this changes a bit the tcl script as well, but I really think it's best to make this match more reliable.

**Review by paulmnt — CHANGES_REQUESTED** [review]

File: utils/sldgen/sld_generate.py (hunk 1 above)
Comment (davide-giri) [reply utils/sldgen/sld_generate.py:1575]: Adding the 32 is a good idea. In addition, how about using `re.fullmatch` and removing the `re.I` flag? That would be that we are only matching full words enclosed by underscore characters: `_fl32_`, `_fl32out_`, `_fl32in_`.

Comment (paulmnt) [reply utils/sldgen/sld_generate.py:1575]: Good idea, agreed!

Comment (paulmnt) [reply utils/sldgen/sld_generate.py:1575]: Please make sure this works even when the 'fl32' is at the end of the HLS configuration string though.

Comment (davide-giri) [reply utils/sldgen/sld_generate.py:1575]: It does, I've tested that.

**Review by paulmnt — APPROVED** [review]

**Review by GuyEichler — APPROVED** [review]
The changes look good to me

**Pull request merged**